web/html/src/utils/functions.ts

```diff
@@ -116,16 +116,34 @@ function capitalize(str: string): string {
 }
 
 function generatePassword(): string {
-  const length = Math.floor(Math.random() * 10) + 15;
-  // See https://stackoverflow.com/a/68617567/1470607
-  const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789:-_";
-  const array = new Uint32Array(charset.length);
-  window.crypto.getRandomValues(array);
+  const length = 24; // fixed length for better security
+  const lowercase = "abcdefghijklmnopqrstuvwxyz";
+  const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  const numbers = "0123456789";
+  const specials = ":-_!@#$%^&*()"; // extended set of special characters
+
+  // Helper function to get a random character from a given set
+  const allChars = lowercase + uppercase + numbers + specials;
+
+  // Ensure at least one character from each category
+  const getRandomChar = (charset) => charset[Math.floor(Math.random() * charset.length)];
 
   let password = "";
-  for (let i = 0; i < length; i++) {
-    password += charset[array[i] % charset.length];
+  password += getRandomChar(lowercase);
+  password += getRandomChar(uppercase);
+  password += getRandomChar(numbers);
+  password += getRandomChar(specials);
+
+  const array = new Uint8Array(length - 4); // Remaining characters
+  window.crypto.getRandomValues(array);
+
+  for (let i = 0; i < array.length; i++) {
+    password += allChars[array[i] % allChars.length];
   }
+
+  // Shuffle the password to avoid guaranteed character positions
+  password = password.split('').sort(() => 0.5 - Math.random()).join('');
```

Comment:
For shuffling, you could also use the [Fisher–Yates shuffle](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle).